### ISSUE TITLE
Converting 0-1 length glob results to optional file output

### ISSUE THREAD

**Issue by OgnjenMilicevic** [openwdl/wdl#384]
Hello, 

it could be that there is a very easy solution to this but I am struggling with a specific case. The name of my optional output is discovered at runtime from an input file header, so in advance I only know the suffix, which enables me to glob it. For compatibility, I would like this output to be a File? type, not Array[File?] or anything similar. Still, if I grab it with [0] I get an error if the array is zero-length, I cannot just grab it by name because the value is obtained as a result of a head -1 command. As I understand, I cannot write if conditionals in the output block. An example snippet is below:

command {
    if [ -f ~{ioe} ]; then
      suppa.py psiPerEvent -i ~{ioe} -e ~{abund} -o $(head -1 ~{abund})_event
    fi
  }

  output {
    File? psi_event = glob("*_event.psi")[0]
  }

Notice the name will be $(head -1 ~{abund})_event.psi. Is there a solution to this I am not aware of?

Thanks!

**Comment by rhpvorderman** [comment]
You actually can write if statements in the output block as far as I know.
Also, you can make stuff easier for yourself by echoing the filename to stdout and use stdout as your filename. Provided your python program does not print messages to stdout with print. 
```WDL
command {
  if [ -f ~{ioe} ]; then
    FILENAME=$(head -1 ~{abund})_event
    echo $FILENAME
    suppa.py psiPerEvent -i ~{ioe} -e ~{abund} -o $FILENAME
  fi
}

output {
  File? psi_event =  stdout()
}
```

Hope this helps.

**Comment by illusional** [comment]
@rhpvorderman's solution is great (though I think you'll need to call `read_string(stdout())` - because stdout is already a File Type). To supplement this answer, I think you might have trouble using an `if` ternary block (as only the development spec has `None` literal, otherwise you could do something like:

```wdl
version development

task YourTask {
  # ...
  output {
    File? psi_event =  if (length(glob("*_event.psi")) == 1) then glob("*_event.psi")[0] else None
}
```

**Comment by rhpvorderman** [comment]
@illusional Thanks for catching that mistake. I updated the code example in the post.

The non existence of `None` is indeed a bit annoying in these use cases. But it should be solved in WDL 2.0

**Comment by OgnjenMilicevic** [comment]
Thank you for a prompt response, and two options besides. :)
Closing this issue!

**Comment by OgnjenMilicevic** [comment]
Just to check @illusional, am I correct that in this case we can use:

`File? psi_event =  if (length(glob("*_event.psi")) == 1) then glob("*_event.psi")[0] else "None"`

And it would work even in WDL1.0?

**Comment by illusional** [comment]
In a funny way, it actually will because `"None"` will resolve your an empty file (unless you have a file called None). But my intention was rather the literal `None` - however that's only available in WDL 2.0 (development) version.
